diesel/src/sqlite/types/date_and_time/chrono.rs

```diff
@@ -0,0 +1,263 @@
+extern crate chrono;
+
+use std::error::Error;
+use std::io::Write;
+use self::chrono::{NaiveDateTime, NaiveDate, NaiveTime, Datelike};
+
+use sqlite::Sqlite;
+use sqlite::connection::SqliteValue;
+use types::{Date, FromSql, IsNull, Time, Timestamp, ToSql, Text};
+
+const SQLITE_TIME_FMT: &'static str = "%0H:%0M:%0S%.6f";
+
+fn parse_sqlite_date(date: &str) -> Result<NaiveDate, Box<Error+Send+Sync>> {
+    let negative_year = date.starts_with('-');
+
+    let ymd: Vec<u32> = date.split('-').flat_map(str::parse).collect();
+    if ymd.len() != 3 {
+        return Err("Cannot parse date".into())
```

Comment:
Maybe add a more precise error message here? This is the same as below, so people getting this error and searching for the string will not know which one it was.